pkg/token/token.go

```diff
@@ -226,25 +228,37 @@ func (g generator) FormatJSON(token string) string {
 // Verifier validates tokens by calling STS and returning the associated identity.
 type Verifier interface {
 	Verify(token string) (*Identity, error)
+	ReloadConfig(c config.Config)
 }
 
 type tokenVerifier struct {
+	lock      sync.Mutex
```

Comment:
Could we use a RWMutext here?  I think there will be a lot more reads than writes.

Reply:
No problem.

Reply:
And i see more lock is using `sync.Mutex`, do we need to change? or create another PR to fix it?

https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/143/files#diff-09856fe9becddf0199651f451409356aR398